ocaml/idl/datamodel.ml

```diff
@@ -5232,10 +5261,15 @@ module SDN_controller = struct
         ; field   ~qualifier:StaticRO ~lifecycle
             ~ty:Int "port" ~default_value:(Some (VInt 0L))
             "TCP port of the controller"
+
+        ; field   ~qualifier:RW ~lifecycle:[Prototyped, rel_next, ""]
```

Comment:
I think this needs to have `DynamicRO` and and explicit `set_local_ports` function. With `RW`, you'll get that function generated for you, but it would only set the DB field. You need to override it to set the DB _and_ update the OVS. You have add/remove functions above, but they don't match the field name (which has "ports" not "port"). It may be easier to not have the add/remove functions, but just `set_local_ports`.

Reply:
However, I see now that you've added better typed parameters to the add/remove functions, with a protocol enum. Let's leave it like that, but we still need the field to have `DynamicRO` to avoid the auto-generated setters.

Reply:
Yes, I like this design.